src/main/java/uk/gov/justice/digital/hmpps/csr/api/service/DetailService.kt

```diff
@@ -34,45 +33,37 @@ class DetailService(
 
         log.debug("Fetching modified detail for $planUnit")
         val modifiedDetails = detailRepository.getModifiedDetails(planUnit)
-                .map {
-                    putDetailOnCorrectDateTime(it)
-                    it
-                }
         log.info("Found ${modifiedDetails.size} modified details for $planUnit")
 
-        return DetailDto.from(modifiedShifts + modifiedDetails)
+        return mapToDetailsDto(modifiedShifts + modifiedDetails)
     }
 
     companion object {
         private const val FULL_DAY_ACTIVITY = -2_147_483_648L
-        private const val DAY_IN_SECONDS = 86_400L
         private val log = LoggerFactory.getLogger(DetailService::class.java)
 
-        private fun putDetailOnCorrectDateTime(detail: Detail) {
-            if (detail.startTimeInSeconds != null && detail.startTimeInSeconds!! < 0 && detail.startTimeInSeconds!! > FULL_DAY_ACTIVITY) {
-                detail.shiftDate = detail.shiftDate.minusDays(1)
+        private fun mapToDetailsDto(details: Collection<Detail>): Collection<DetailDto> {
+            return details.map {
+                val detailStart = it.startTimeInSeconds?.let { it1 -> calculateDetailDateTime(it.shiftDate, it1) }
+                val detailEnd = it.endTimeInSeconds?.let { it1 -> calculateDetailDateTime(it.shiftDate, it1) }
+                DetailDto.from(it, detailStart, detailEnd)
             }
-            detail.startTimeInSeconds = detail.startTimeInSeconds?.let { time -> ensureTimeWithinDayBounds(time) }
-            detail.endTimeInSeconds = detail.endTimeInSeconds?.let { time -> ensureTimeWithinDayBounds(time) }
         }
 
-        // We shouldn't send times outside of what is supported by LocalTime.ofSecondOfDay()
-        private fun ensureTimeWithinDayBounds(time: Long): Long {
+        private fun calculateDetailDateTime(shiftDate: LocalDate, detailTime: Long): LocalDateTime {
+            val shiftDateTime = shiftDate.atStartOfDay()
+            // If the start value is a negative number the shift detail actually starts the previous day
             return when {
-                time == FULL_DAY_ACTIVITY -> {
-                    0
+                detailTime in (FULL_DAY_ACTIVITY + 1)..-1 -> {
+                    shiftDateTime.minusSeconds(abs(detailTime))
```

Comment:
Personally not a fan of func invocation inside function invocation

Reply:
I'm not sure I like it, its creating a range so we're saying 
If detail time is between -23234234 and -1
effectively 
if(detailTime > -23133442 && detailTime < 0) 
which I think I prefer.